be/src/storage/rowset/segment_group.h

```diff
@@ -0,0 +1,130 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#pragma once
+
+#include "storage/rowset/segment.h"
+
+namespace starrocks {
+
+class ShortKeyIndexGroupIterator;
+class ShortKeyIndexDecoderGroup;
+class SegmentGroup;
+using SegmentGroupPtr = std::unique_ptr<SegmentGroup>;
+
+class ShortKeyIndexGroupIterator {
+public:
+    using iterator_category = std::random_access_iterator_tag;
+    using value_type = Slice;
+    using pointer = Slice*;
+    using reference = Slice&;
+    using difference_type = ssize_t;
+
+    ShortKeyIndexGroupIterator() : _decoder_group(nullptr), _ordinal(0) {}
+
+    ShortKeyIndexGroupIterator(const ShortKeyIndexDecoderGroup* decoder_group, ssize_t ordinal);
+
+    ShortKeyIndexGroupIterator& operator-=(ssize_t step) {
+        _ordinal -= step;
+        return *this;
+    }
+
+    ShortKeyIndexGroupIterator& operator+=(ssize_t step) {
+        _ordinal += step;
+        return *this;
+    }
+
+    ShortKeyIndexGroupIterator& operator++() {
+        ++_ordinal;
+        return *this;
+    }
+
+    ShortKeyIndexGroupIterator& operator--() {
+        --_ordinal;
+        return *this;
+    }
+
+    bool operator!=(const ShortKeyIndexGroupIterator& rhs) const {
+        return _ordinal != rhs._ordinal || _decoder_group != rhs._decoder_group;
+    }
+
+    bool operator==(const ShortKeyIndexGroupIterator& rhs) const {
+        return _ordinal == rhs._ordinal && _decoder_group == rhs._decoder_group;
+    }
+
+    ssize_t operator-(const ShortKeyIndexGroupIterator& rhs) const { return _ordinal - rhs._ordinal; }
+
+    bool valid() const;
+
+    Slice operator*() const;
+
+    ssize_t ordinal() const { return _ordinal; }
+
+private:
+    const ShortKeyIndexDecoderGroup* _decoder_group;
+    ssize_t _ordinal;
+};
+
+class ShortKeyIndexDecoderGroup {
```

Comment:
DecoderGroup sounds a little strange

Reply:
This class contains `ShortKeyIndexDecoder` for each segment, and perform the binary search logic. Any better idea?